jupyterlab_git/git.py

```diff
@@ -286,6 +287,21 @@ async def clone(self, path, repo_url, auth=None):
             if auth.get("cache_credentials"):
                 await self.ensure_credential_helper(path)
             env["GIT_TERMINAL_PROMPT"] = "1"
+
+            if meta_data_checked:
```

Comment:
As there is already a condition on `auth` (aka are credentials provided or not?), I would do the test on the new flag first. And depending on its value, I would pick the correct command list.

Then after cloning (either with or without authentication), I will do a test against the flag again to know if the `.git` repository needs to be removed.